src/components/Organisms/TodoCard/index.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import AddTaskButton from "../../Atoms/AddTaskButton/index.jsx";
+import Task from "../../Molecules/Task/index.jsx";
+import COLOR from "../../../variables/color.js";
+
+const TodoCard = () => {
+  const [taskList, settaskList] = useState();
+
+  const onAddTaskButtonClick = () => {
+    settaskList((currentList) => {
+      if (currentList !== undefined) {
+        return [...currentList, { name: "", initializing: true }];
+      } else {
+        return [{ name: "", initializing: true }];
+      }
+    });
+  };
+  const onTaskComplete = (index) => {
+    settaskList((currentList) => {
+      const nextList = currentList.filter((_, idx) => index !== idx);
+      return nextList;
+    });
+  };
+  const onTaskNameChange = (value, index) => {
+    if (!value) {
+      onTaskComplete(index);
+    } else {
+      settaskList((currentList) => currentList.splice(index, 1, value));
```

Comment:
```suggestion
      currentList.splice(index, 1, value)
      settaskList((currentList) => [...currentList]);
```
spliceは、mapやfilterと違って、**その場で**配列操作を行います。
つまり、mapやfilterは処理を行った結果の新しい配列を返り値として返しますが、
spliceは操作対象の配列自身（今回であれば `currentList`）を改変するということです。

そして、spliceの返り値は取り除かれた要素を含む配列です。
指定した要素を取り除いた結果ではなく、取り除かれた要素の方を返すのです。
そのため、現状のコードでは、名前の変更を行った瞬間に変更前のタスク名が一つだけ存在する状態に変化すると思います。

上記の書き方をすればこの問題は回避できると思いますが、個人的にはmapを使うやり方でもいいかなと思っています。（sortやspliceのようなその場で操作するメソッドはReactと相性が悪いのが理由です。）

参考：　https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/splice